osu.Desktop/DiscordRichPresence.cs

```diff
@@ -96,6 +98,21 @@ private void updateStatus()
             client.SetPresence(presence);
         }
 
+        private string truncate(ReadOnlySpan<char> str)
+        {
+            if (Encoding.UTF8.GetByteCount(str) <= 128)
+                return new string(str);
```

Comment:
The input should be a string to avoid extra allocation when not truncating. Use a `ReadOnlyMemory<char>` to truncate.

Reply:
ReadOnlyMemory gets ugly. You lose any performance benefits as soon as you call `.Span` on it,`GetByteCount()` doesn't support `ReadOnlyMemory<>`, and you have to decorate callsites with `.AsMemory()`.

Reply:
`ReadOnlyMemory<T>.Span` has 128 bytes in asm. Really surprises me.
Since `GetByteCount` supports string, it should be better assigning the span after the early return route.

Reply:
Fyi the `.Span` inside the loop will cause the same runaway condition. Here's a benchmark I've done:

|         Method | Length |        Mean |      Error |     StdDev | Gen 0/1k Op | Gen 1/1k Op | Gen 2/1k Op | Allocated Memory/Op |
|--------------- |------- |------------:|-----------:|-----------:|------------:|------------:|------------:|--------------------:|
|   TruncateSpan |      1 |    23.66 ns |  0.2237 ns |  0.2093 ns |      0.0076 |           - |           - |                24 B |
| TruncateMemory |      1 |    17.61 ns |  0.2475 ns |  0.2316 ns |           - |           - |           - |                   - |
|   TruncateSpan |    128 |    51.66 ns |  1.0912 ns |  1.7929 ns |      0.0892 |           - |           - |               280 B |
| TruncateMemory |    128 |    25.07 ns |  0.2214 ns |  0.2071 ns |           - |           - |           - |                   - |
|   TruncateSpan |    200 | 1,858.51 ns | 16.5747 ns | 15.5040 ns |      0.1755 |           - |           - |               552 B |
| TruncateMemory |    200 | 1,981.22 ns | 22.4968 ns | 19.9428 ns |      0.0877 |           - |           - |               280 B |